src/tiledimage.js

```diff
@@ -353,10 +373,7 @@ $.extend($.TiledImage.prototype, $.EventSource.prototype, /** @lends OpenSeadrag
      */
     destroy: function() {
         this.reset();
-
-        if (this.source.destroy) {
-            this.source.destroy(this.viewer);
-        }
```

Comment:
No. a) it was not even documented. b) it was hacky way for one tile source implementation to free its data. c) it is now automatic. If someone was so crafty they noticed support for this, they will be likewise crafty to update to the newest API. I had problems noticing support for this at all.

Reply:
> If someone was so crafty they noticed support for this, they will be likewise crafty to update to the newest API.

There's a lot of legacy code out there where the original crafty developer is no longer involved (and often there are no more developers involved). I guess my gut feeling would be to keep the guard, and possibly add a warning or error message if `this.source.destroy` doesn't exist, but I don't feel too strongly... Thoughts, @iangilman?

Reply:
> c) it is now automatic.

What do you mean by "automatic"? Is it enforced elsewhere in the code (adding it to an implementation that doesn't have one)? Users can pass in a variety of things to specify a tilesource, I just want to make sure all bases are covered.

Reply:
Similarly to drawers, TileSource cannot just decide it wants to keep internally some data for rendering. All data comes through cache and the system governs when it is freed. Like reset(). Whether it is being turned into a zombie, freed immediately (and what format, e.g. a texture/canvas... Anything) is not a TileSource concern.

Reply:
The base TileSource now has a destroy:

https://github.com/openseadragon/openseadragon/pull/2407/files#diff-de71b1e15703e016afd7f4072b823467fbc11e6348bc2413de60ce1f75739320R645

… So I think that covers this.

Reply:
@iangilman it is true that the base `TileSource` has this; my concern was that there has been a code path where older implementations could be passed in and assumed to be correct, for example: 

https://github.com/openseadragon/openseadragon/blob/04211fbb27928251bca67cf00a5fae683f8c9880/src/viewer.js#L2701-L2704

Of course, this is likely to break things for other reasons too. I'm fine with ditching that guard if you're good with it.

Reply:
Yeah, I don't think we explicitly support that... if you make a custom tile source, you should base it on our base TileSource. That said, having the guard in is easy enough and doesn't harm anything. I guess I'm fine either way. Thank you for spotting it though!

Reply:
Solved by imposing force-inheritance of a TileSource prototype.

Reply:
Hmm I removed the tile source prototype extension, since it does not work well. IMHO if tileSource does not comply with the interface, it is up to the devs to fix the compliance. For example, the WebglDrawer also calls `isTainted()` without checking whether this new property exists, so this is the same case.

Reply:
IMHO the tileSource creation routine is also pretty tangled, I made issue about that, and it would deserve a refactoring. See https://github.com/openseadragon/openseadragon/issues/2474